components/layout_2020/flow/mod.rs

```diff
@@ -755,11 +708,44 @@ fn solve_inline_margins_for_in_flow_block_level(
     }
 }
 
-fn clamp_between_extremums(mut size: Length, min_size: Length, max_size: Option<Length>) -> Length {
-    if let Some(max_size) = max_size {
-        if size > max_size {
-            size = max_size;
-        }
+fn clamp_between_extremums(size: Length, min_size: Length, max_size: Option<Length>) -> Length {
+    clamp_below_max(size, max_size).max(min_size)
+}
+
+fn clamp_below_max(size: Length, max_size: Option<Length>) -> Length {
+    max_size.filter(|max_size| size > *max_size).unwrap_or(size)
```

Comment:
This feels a bit clever to me, as in needs a few more seconds to follow what’s going on. What do you think of using `.min(…)`?

Reply:
All I needed is <s>love</s> `map` and `min` indeed.